include/remill/Arch/Instruction.h

```diff
@@ -254,6 +257,148 @@ class Instruction {
     kCategoryConditionalAsyncHyperCall,
   } category;
 
+
+  class Flow {};
+
+  class DirectFlow : Flow {
+   public:
+    DirectFlow() = delete;
+    DirectFlow(uint64_t known_target, DecodingContext static_context);
+
+    uint64_t known_target;
+    DecodingContext static_context;
+
+    bool operator==(const DirectFlow &rhs) const;
+  };
+
+  class IndirectFlow : Flow {
+   public:
+    // We may have info in the decoder that tells us a context value
+    std::optional<DecodingContext> maybe_context;
+
+    IndirectFlow() = delete;
+    IndirectFlow(std::optional<DecodingContext> maybe_context);
+
+    bool operator==(const IndirectFlow &rhs) const;
+  };
+
+  class FallthroughFlow : Flow {
+   public:
+    FallthroughFlow(DecodingContext fallthrough_context);
+    FallthroughFlow() = delete;
+
+    DecodingContext fallthrough_context;
```

Comment:
We should know the next address right? Why is it not similar to `DirectFlow`?

Reply:
Fallthroughs "must" go to next_pc which is defined as the next instruction, otherwise it's not a fallthrough